### PR TITLE
Migrated to System.Text.Json

### PR DESCRIPTION
**Pull request by Silvenga**
The https://github.com/kubernetes-client/csharp project decided to migrate to System.Text.Json over the mature JSON.NET library. To maintain compatibility and to continue to receive bug fixes from upstream, we must also migrate to the same.

This PR migrates to System.Text.Json in every case except when calculating JSON Diffs (the library has not moved to System.Text.Json). A hack was introduced to share the same naming conventions/type converters from upstream kubernetes-client. Additionally, the YAML serializer naming conventions are now shared with kubernetes-client.

This is a breaking change:
- Existing projects must migrate from using JsonPropertyAttributes to JsonPropertyNameAttributes.
- The YAML serializer is also shared with kubernetes-client and is no longer overridable via injection.
- Microsoft.Rest.HttpOperationException no longer exists, any place this is used must move to k8s.Autorest.HttpOperationException.

Resolves https://github.com/buehler/dotnet-operator-sdk/issues/434.

### CODE REVIEW

**Comment by Silvenga** [comment]
Large foundational changes, I want to test on my side a bit before merging.

**Comment by Silvenga** [comment]
Tested, looks good!

**Comment by buehler** [comment]
btw: funny branch name 😂

File: src/KubeOps/Operator/Webhooks/AdmissionRequest.cs [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
     public string Operation { get; init; } = string.Empty;
 
+    [JsonPropertyName("object")]
     public TEntity? Object { get; set; }
 
+    [JsonPropertyName("oldObject")]
     public TEntity? OldObject { get; set; }
 
+    [JsonPropertyName("dryRun")]
     public bool DryRun { get; set; }
```
[the remark this answers is not in the document]
Comment (Silvenga) [reply src/KubeOps/Operator/Webhooks/AdmissionRequest.cs:20]: You make a good point, I'm not sure why I needed to set these, the default settings are (from the K8s project):

```
KubernetesJson.JsonSerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull;
KubernetesJson.JsonSerializerOptions.PropertyNamingPolicy = JsonNamingPolicy.CamelCase;
KubernetesJson.JsonSerializerOptions.Converters.Add((JsonConverter) new KubernetesJson.Iso8601TimeSpanConverter());
KubernetesJson.JsonSerializerOptions.Converters.Add((JsonConverter) new KubernetesJson.KubernetesDateTimeConverter());
KubernetesJson.JsonSerializerOptions.Converters.Add((JsonConverter) new KubernetesJson.KubernetesDateTimeOffsetConverter());
KubernetesJson.JsonSerializerOptions.Converters.Add((JsonConverter) new V1Status.V1StatusObjectViewConverter());
KubernetesJson.JsonSerializerOptions.Converters.Add((JsonConverter) new JsonStringEnumConverter());
```

Which includes the camel case convention. But I couldn't patch incoming manifests without setting the JsonPropertyName's. I didn't need to modify anything regarding `CustomKubernetesEntity`.

[the remark this answers is not in the document]
Comment (Silvenga) [reply src/KubeOps/Operator/Webhooks/AdmissionRequest.cs:20]: Ugh, it's the way `JsonSerializer.Deserialize` works against `JsonElement`. The `DotnetKubernetesClient` project needs to be updated to use the K8s conventions, it's currently using. I'll make a pr.

[the remark this answers is not in the document]
Comment (Silvenga) [reply src/KubeOps/Operator/Webhooks/AdmissionRequest.cs:20]: Should be fixed in https://github.com/buehler/dotnet-kubernetes-client/pull/125.

[the remark this answers is not in the document]
Comment (buehler) [reply src/KubeOps/Operator/Webhooks/AdmissionRequest.cs:20]: oh, wow. Didn't catch that.
This problem with the "upstream" of the kubernetes client is one of the reasons why I'd like to remove the reference to "dotnet kubernetes client" and archive the repository.

Maybe it is an idea to directly include the client into this repository. My original idea for the client was to be used in other projects that need that generic approach. I definitely need to investigate that further.

[the remark this answers is not in the document]
Comment (buehler) [reply src/KubeOps/Operator/Webhooks/AdmissionRequest.cs:20]: Much appreciated! Let us keep the generic version then. But I'd like to wait with the breaking change until it's necessary to upgrade. I think about merging it into a next branch to use it as prerelease version. Do you think the upgrade is required now (or rather short term)?

[the remark this answers is not in the document]
Comment (Silvenga) [reply src/KubeOps/Operator/Webhooks/AdmissionRequest.cs:20]: Before I forget, I can remove the commit that adds the JsonPropertyName, I can go either way. My testing branch has them removed now.

But yeah, I think this is a good long term upgrade (pre-upgrade to v8, which includes even more breaking changes). I know this indirectly fixes #434 (which might only impact my project 😸) - but I'm now submoduling my fork (and trying to cherry-pick back into pr's when I make a new change).

**Comment by buehler** [comment]
Alright @Silvenga 
You can use the pre release version https://github.com/buehler/dotnet-operator-sdk/releases/tag/v7.0.0-prerelease.1
:) as such you don't need to use the fork anymore
I try to keep the next branch up with changes to the master

**Comment by Silvenga** [comment]
Woot! Thanks @buehler.

**Pull request merged**